aten/src/ATen/NamedTensor.h

```diff
@@ -47,6 +47,26 @@ struct CAFFE2_API NamedTensorMeta : public c10::NamedTensorMetaInterface {
   std::vector<Dimname> names_;
 };
 
+// When NamesMode is disabled, then all operations ignore tensors' names fields.
+// Concretely speaking, all tensors are treated as having nullopt names.
+struct CAFFE2_API NamesMode {
+  static bool is_enabled();
+  static void set_enabled(bool enabled);
+};
+
+// A RAII, thread local (!) guard that enables or disables names upon
+// construction, and sets it back to the original value upon destruction.
```

Comment:
might be helpful to also describe why this is needed

Reply:
That's good to have in general. I'm not sure what the guidelines for "how to best use NoNamesGuard" is right now, so I didn't write that in.